<launch>
  <arg name="team" default="1"/>
 
  <group ns="robot$(arg team)">
	<!-- Robot drivers -->
	<node pkg="rosaria" name="robot$(arg team)" type="RosAria">
 		<param name="port" value="/dev/pioneer"/>
    </node>

	<!-- USB webcam -->
	<node pkg="usb_cam" name="front_camera" type="usb_cam_node">
		<!-- TODO: Check port on shuttle xpc! -->
		<param name="video_device" value="/dev/video0"/>

		<!-- parameters can be changed -->
	    <!--<param name="image_width" value="1024"/>
     		<param name="image_height" value="768"/>
	    	<param name="autofocus" value="true"/>-->
    </node>

	<!-- RPlidar laser scanner -->
	<node pkg="rplidar_ros" name="front_laser" type="rplidarNode" output="screen">
	  <param name="serial_port"         type="string" value="/dev/rplidar"/>  
	  <param name="serial_baudrate"     type="int"    value="115200"/>
	  <param name="frame_id"            type="string" value="laser"/>
	  <param name="inverted"            type="bool"   value="false"/>
	  <param name="angle_compensate"    type="bool"   value="true"/>
	  <remap from="scan" to="front_laser/scan"/>
    </node>

	<!-- KI node controlling the robot -->
	<include file="$(find player)/launch/player.launch">
		<arg name="team" value="$(arg team)" />
	</include>
  </group>

</launch>
